### ISSUE TITLE
Can't get it working

### ISSUE THREAD

**Issue by alejandroalis** [andi38/TPMS#5]
Hi!

I've tried running the example on an ESP32 with the 433MHz wheel sensors, but no luck. Communication with the CC1101 is working correctly; it initializes everything, but it doesn’t receive any data. I know the TPMS is sending data because I can read it correctly with the RTL-SDR, and thanks to the rtl_433 utility, I can see readings compatible with the ‘truck’ sensor type. I tried changing the sync bytes to other values based on the readings I had from the SDR, but still nothing. I noticed that when compiling the code, there was an error (a missing ; on line 81 and the setFlag function isn’t declared at the beginning), so I'm wondering if the code was really tested or if you made any changes. Any help would be appreciated!

**Comment by unverbuggt** [comment]
Hi,

I'm currently working on reading [tpms_ford.c](https://github.com/merbanan/rtl_433/blob/master/src/devices/tpms_ford.c) devices and I've learned something yesterday: The TPMS sensors send using OOK modulation when triggered by the 250kHz trigger tool and they use FSK while the wheels are spinning.
This is not a big problem for a SDR (Maybe it is possible for rtl_433 to show the used modulation?), but a big problem for the CC1101 chip. You can try adding `radio.setOOK(true);` in the setup routine. But best is to use Universal Radio Hacker with the SDR and record the signal and then check the modulation.

**Comment by alejandroalis** [comment]
Hello unverbuggt! I’m using some from Aliexpress for a motorcycle; they only have a pressure sensor, and I noticed they detect when the vehicle moves by changes in the pressure, without an actual motion sensor—pretty clever. Using the RTL-SDR, I can capture the data, but no configuration works with the CC1101. I tried following your example and changing everything, but still nothing… did you ever manage to get it working?

**Comment by unverbuggt** [comment]
yes, I got it working. I'll fork this repo and share my code in the next days.

Main problem with receiving these signals is the short preamble (the chip needs 16 bits). Also the preamble isn't manchester encoded, which, as I understand it, means we cannot use the manchester decoding on the chip (I don't know how @andi38 got it working, also with the one byte sync word...).
The other problem is that I currently only get stable results if I use the sensor ID as sync word, which means that If sensors start with a different ID, they cannot be received (I got two Ford cars with different generations of sensors, but also with different tyre sizes, so I wouldn't need to mix these).
I'm planning of writing an article about this, but this is going to take some time.

**Comment by alejandroalis** [comment]
The truth is I spent an entire day on it and couldn’t get it to work. I’m developing a specific tablet for motorcycles from scratch (I’m an electronics engineer) and, honestly, doing it all without help... I hope to launch a Kickstarter in a couple of months. Being able to read directly from standard TPMS sensors would be incredibly useful. If you could give me a hand and share the code that works for you, it would be a huge help—I’m going crazy with everything!... you can check it on www.alsenms.com

**Comment by unverbuggt** [comment]
I've forked [here](https://github.com/unverbuggt/TPMS)

**Comment by alejandroalis** [comment]
Sorry for not replying sooner... I'm in the area affected by the DANA in Spain, and it's been complete chaos. I'll try to get back to this and test the code you shared. Thank you! I'll let you know once I've tried it. :)

**Comment by alejandroalis** [comment]
Thanks, I’ve tested it, and after analyzing the data, I was able to extract the pressure since the sensors don’t seem to be the "Truck" model but rather a different one. Thanks for the help!

**Comment by unverbuggt** [comment]
Good to hear. You're welcome.

**Comment by denbon** [comment]
> Спасибо, я проверил это, и после анализа данных мне удалось извлечь давление, поскольку датчики, похоже, не модели "Truck", а какой-то другой. Спасибо за помощь!

alejandroalis How can I contact you? Write your email. I have a couple of questions for you.

**Comment by denbon** [comment]
> Thanks, I’ve tested it, and after analyzing the data, I was able to extract the pressure since the sensors don’t seem to be the "Truck" model but rather a different one. Thanks for the help!

alejandroalis How can I contact you? Write your email. I have a couple of questions for you.

**Comment by alejandroalis** [comment]
hi denbon, what's your question?

**Comment by denbon** [comment]
> hi denbon, what's your question?

I ran into a problem setting up the ambient light sensor in Android on a Rockchip chip, and I found on Stack Overflow that you asked a similar question at the link:
https://stackoverflow.com/questions/77099489/kernel-driver-not-detected-by-apps-in-android
I have a similar problem, could you tell me how you solved this problem, that's why I wanted to get your email to contact you.
Please, be kind enough to help me, with respect to you.

**Comment by alejandroalis** [comment]
I thought it was a question about the TPMS project, haha. Ok, if it's about the light sensor on Android, it was about a year ago, but in the end, I couldn't get it to work and dedicated the time to other aspects of the project. The documentation on Android is terrible, and some things, like that one, seem simple, but no matter how much you look, you can't see the problem.

**Comment by denbon** [comment]
Ok, thanks for the answer. Unfortunately, I agree with you that the chip manufacturer Rockchip is negligent about the software, and there is not even normal documentation. They provide raw Android source code.

**Comment by alejandroalis** [comment]
About this, I'm thinking about buying a Mediatek board and its SDK. The SOM is much better and it has LTE, WiFi, BT... all included. The CPU is better... but... the SDK costs 3000 USD... although I think it should be very well documented. Also, the SOM price is cheaper than Rockchip (including LTE, WiFi...)

[mtk6785 user manual OBOLA.pdf](https://github.com/user-attachments/files/18948083/mtk6785.user.manual.OBOLA.pdf)

**Comment by denbon** [comment]
Only the price is expensive. But it turns out that in a cheap board, there is no documentation or normal Android source code. It turns out to be just hardware. But 3000 dollars. this is of course a lot, not everyone can afford to buy it. It turns out that the most affordable board is raspberry pi 5, it has good support.

**Comment by alejandroalis** [comment]
I want it for a commercial product, and the Raspberry Pi won't work for that. For commercial development, 3000 dollars is not much money, so I'm considering buying it. However, I would need to know how good the documentation is before making the decision.

**Comment by HCA1** [comment]
Hey @unverbuggt and @alejandroalis , Im currently doing this as a final year project which will. be due soon, I'm stuck, I got the EV6T sensors, an esp32 room, an EL-50448 to trigger the sensors and they are not in a tire, I'm using a cc1101 module, I must say, I haven't made much progress on this project, it's not transmitting, Any help on this would be very much appreciated please, can I contact you through any other channel you would prefer

**Comment by alejandroalis** [comment]
Hey, there’s actually a fork with working code that I’ve tested myself: [https://github.com/unverbuggt/TPMS](https://github.com/unverbuggt/TPMS)

If you follow any of the common ESP32 + CC1101 tutorials online, you should be able to get it working. If not, double-check your wiring and try to follow a tutorial that uses exactly the same hardware. Once you get that part working, the code from the link above should work without issues.

**Comment by unverbuggt** [comment]
Best make sure that the sensor is working first with a sdr (f.ex. rtl-sdr) and the [rtl_433 Tool](https://github.com/merbanan/rtl_433).

the sensor should transmit three times OOK and then three times FSK. While driving only FSK...

**Comment by HCA1** [comment]
Thanks for your help, I’m getting data now.
